R/cancelOrders.R

```diff
@@ -97,27 +98,64 @@
 #'
 
 cancelOrders <-
-  function(marketId, betIds = NULL, sizeReductions = NULL, suppress = FALSE, sslVerify = TRUE) {
+  function(marketId = NULL, betIds = NULL, sizeReductions = NULL, customerRef = NULL, suppress = FALSE, sslVerify = TRUE) {
 
     options(stringsAsFactors = FALSE)
 
-    if (is.null(sizeReductions))
-      sizeReductions = rep("NULL", length(betIds))
-
     if (length(betIds) != length(sizeReductions))
 
       return("Bet ID and Size Reduction vectors need to have the same length")
 
-    cancelOrderOps <-
-      paste0(
-        '[{"jsonrpc": "2.0","method": "SportsAPING/v1.0/cancelOrders","params":{"marketId": "',marketId,'","instructions": [',
-        paste0(sapply(as.data.frame(t(data.frame(
-          betIds, sizeReductions
-        ))),function(x)
-          paste0('{"betId":"',x[1],'","sizeReduction":"',x[2],'"}')),collapse = ","),']},"id": "1"}]'
-      )
+    cancelOrdersOps <-
+      data.frame(jsonrpc = "2.0", method = "SportsAPING/v1.0/cancelOrders", id = 1)
+
+    #required fields
+    cancelOrdersOps$params <-
+      data.frame(marketId = marketId)
+    #optional fields
+    if(!is.null(customerRef)){
+      if(!is.null(marketId)){
+        cbind(cancelOrdersOps$params, customerRef = customerRef)
+      } else {
+        cancelOrdersOps$params$customerRef <- customerRef
+      }
+    }
+
+    #required fields
+    if(!(is.null(betIds)&&is.null(sizeReductions))){
+      if(!is.null(betIds)){
+        insts <-
+        data.frame(
+          betId = betIds)
+        if(!is.null(sizeReductions)){
+          insts <- cbind(insts, sizeReduction = sizeReductions)
+        }
+      } else {
+        insts <-
+          data.frame(
+            sizeReduction = sizeReductions)
+      }
+      #optional fields
+      if(!is.null(betIds)){
+        cbind(insts, betId = betIds)
+      }
+      if(!is.null(sizeReductions)){
+        cbind(insts, sizeReduction = sizeReductions)
+      }
+
+      cancelOrdersOps$params$instructions <-
+        list(insts)
+    }
 
-    cancelOrderOps = gsub("NULL","",cancelOrderOps)
+
+    show(cancelOrdersOps)
+
+    cancelOrdersOps <-
+      cancelOrdersOps[c("jsonrpc", "method", "params", "id")]
+
+    cancelOrdersOps <- jsonlite::toJSON(jsonlite::unbox(cancelOrdersOps))
+
+    show(cancelOrdersOps)
```

Comment:
I don't think the show() is needed here and may be an artefact from your testing